### PR TITLE
fix: Bump Account Nonce Post Regolith

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Post-regolith, account nonces must be incremented for deposit transactions if they are contract creations even if the transaction halts.

### CODE REVIEW

File: crates/revm/src/evm_impl.rs [hunk 1]
```diff
@@ -409,7 +411,26 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
...
             },
-            SuccessOrHalt::Halt(reason) => ExecutionResult::Halt { reason, gas_used },
+            SuccessOrHalt::Halt(reason) => {
+                // Post-regolith, if the transaction is a deposit transaction and the
+                // output is a contract creation, increment the account nonce even if
+                // the transaction halts.
+                #[cfg(feature = "optimism")]
+                {
+                    let is_creation = matches!(output, Output::Create(_, _));
+                    let regolith_enabled = RegolithSpec::enabled(self.data.env.cfg.spec_id);
```
Comment (clabby) [inline crates/revm/src/evm_impl.rs:421]: This looks incorrect - should be using `GSPEC::enabled(REGOLITH)` here I believe.

**Review by clabby — CHANGES_REQUESTED** [review]
LGTM apart from checking if regolith is active

File: crates/revm/src/evm_impl.rs (hunk 1 above)
Comment (refcell) [reply crates/revm/src/evm_impl.rs:421]: Ah shoot, great catch

**Review by clabby — APPROVED** [review]
lgtm 👑

**Pull request merged**